explorer-api/src/Explorer/Web/Api/Legacy/Util.hs

```diff
@@ -72,11 +76,33 @@ collapseTxGroup xs =
     [] -> error "collapseTxGroup: groupOn produced [] on non-empty list (impossible)"
     (x:_) -> (fst x, map snd xs)
 
-decodeTextAddress :: Text -> Either ExplorerError Address
+decodeTextAddress :: Text -> Either ExplorerError (Addr TPraosStandardCrypto)
 decodeTextAddress txt =
-  first (const . Internal $ "Unable to decode address " <> txt <> ".")
-    $ decodeAddressBase58 txt
-
+    case tryBech32 <|> tryBase58 of
```

Comment:
Just noting that, with `cardano-api`, this could be done similarly without having to worry about the address encoding specifics: `deserialiseAddress AsShelleyAddress txt <|> deserialiseAddress AsByronAddress txt`.

Reply:
I do actually disagree with this however, the user-facing encoding has nothing to do with the underlying address format. By **convention**, we use base58 for Byron and bech32 for Shelley, but a Byron addresses encoded in Bech32 would be just fine in principle. I mostly extracted this from the wallet, which actually has an extra `tryBase16` branch as well, to be even more lenient on the address input encoding. I think it'd make sense here to actually.

Reply:
see: https://github.com/input-output-hk/cardano-rest/commit/06f0405cd9ef3c50bb18316ae5415ac6fbba9e80

Reply:
I believe `deserialiseAddress AsShelleyAddress` also supports deserialising Bech32-encoded Byron addresses with the prefix `addr_bootstrap`.

Reply:
But yeah, falling back to Base16 is also a good idea.